src/Carbon/Lang/oc.php

```diff
@@ -86,7 +86,7 @@
         $ordinal = [1 => 'èr', 2 => 'nd'][(int) $number] ?? 'en';
 
         // feminine for year, week, hour, minute, second
-        if (preg_match('/^[yYwWhHgGis]$/', $period)) {
+        if (preg_match('/^[yYwWhHgGis]$/', $period ?? '')) {
```

Comment:
I think we could just skip the `preg_match` call if `$period` is falsy.
```suggestion
        if ($period && preg_match('/^[yYwWhHgGis]$/', $period)) {
```

Reply:
I've added type hint to this closure, is it OK like that, or do you want this check to be there anyway?

Reply:
No it's fine like this, `':period' => (string) $period,` to ensure the string type for text interpolations make complete sense.